src/nvim/eval/vars.c

```diff
@@ -153,6 +184,20 @@ static char *eval_all_expr_in_str(char *str)
   return ga.ga_data;
 }
 
+/// List Vim variables.
+static void list_vim_vars(int *first)
+{
+  list_hashtable_vars(&vimvarht, "v:", false, first);
+}
+
+/// List script-local variables, if there is a script.
+static void list_script_vars(int *first)
+{
+  if (current_sctx.sc_sid > 0 && current_sctx.sc_sid <= script_items.ga_len) {
+    list_hashtable_vars(&SCRIPT_VARS(current_sctx.sc_sid), "s:", false, first);
+  }
+}
+
```

Comment:
`eval_one_expr_in_str()` and `eval_all_expr_in_str()` should come after these two function and before `heredoc_get()`.